test/common/network/listener_impl_test.cc

```diff
@@ -412,13 +412,11 @@ TEST_P(TcpListenerImplTest, SetListenerRejectFractionIntermediate) {
   {
     testing::InSequence s1;
     EXPECT_CALL(random_generator, random()).WillOnce(Return(std::numeric_limits<uint64_t>::max()));
-    EXPECT_CALL(listener_callbacks, onAccept_(_));
+    EXPECT_CALL(listener_callbacks, onAccept_(_)).WillOnce([&] { dispatcher_->exit(); });
```

Comment:
Can you explain this change? It's not clear to me why this is better than the old version.

Reply:
making the dispatcher exit on the connected event makes the test racy, the dispatcher could exit before there is a chance to invoke the tcp listener `onAccept` callback which is effectively the last thing that happens when the connection is established/accepted

Reply:
Oh I see. I thought the events happened in the opposite order. Sounds good.